sql/sql_optimizer.cc

```diff
@@ -2559,7 +2579,8 @@ void JOIN::adjust_access_methods() {
         // From table scan to index scan, thus filter effect needs no recalc.
       }
     } else if (tab->type() == JT_REF) {
-      if (can_switch_from_ref_to_range(thd, tab, ORDER_NOT_RELEVANT, false)) {
+      const char *switch_reason = can_switch_from_ref_to_range(thd, tab, ORDER_NOT_RELEVANT, false);
+      if (switch_reason) {
```

Comment:
Avoid implicit conversions
```suggestion
      if (switch_reason != nullptr) {
```